Rewrite FullScreenTable as an ES Module

kr8n3r opened the pull request:
We want to migrate all of our legacy Javascript files to ESM, so this is the latest in the series of conversions.

Rewrite follows the same conventions we previously used for RadioSelect, FocusBanner, Collapsible checkboxes, ColourPreview, FileUpload, Autofocus, PreviewPane, CopyToClipboard and others.

Added bonus: removes the use of jQuery.

In this PR we:

- rewrite the module as ES Module
- update the tests to be ES Module and polyfill some Javascript methods that JSDOM doesn't return correct values
- remove old module JS file
- remove old module from the legacy bundle build

## How to test

- this is module is used on the scheduling of notifications on the "send" page and
- upload CSV emergency list preview page

tombye reviewed (changes requested):
Really nice work, and so much better to read than the original (I kept thinking this when reviewing ⭐). I did find at least one thing that needs changing but nothing too big.

This review has taken me ages, so sorry for that, but the stuff about changes to `ScreenMock` did need some thinking.

kr8n3r committed:
Add box-sizing:border-box to scrollable table header

Now, that we are no longer using jQuery and its
.width method to get elements content width and
are using vanilla Javascript's
`getBoundingClientRect()`, we get different values
as getBoundingClientRect returns the value
based on the CSS `box-sizing` property of the
element.

By setting `box-sizing: border-box`, the returned
values will include padding and borders.

This resolved the gap between the fixed table
column and the rest of the scrollable table columns.

Found this approach to be most reliable
for both narrow and wide tables across
the browsers our Javascript supports.

Alternative approaches using clientWidth,
offsetWidth and getBoundingClientRect().width
without setting box-sizing produced wider column
for narrow tables even with substracting padding
and border widths from the calculated width.

kr8n3r committed:
Rewrite FullScreenTable as an ES Module

We want to migrate all of our legacy Javascript
files to ESM, so this is the latest in the
series of conversions.

Rewrite follows the same conventions we
previously used for RadioSelect, FocusBanner,
Collapsible checkboxes, ColourPreview, FileUpload,
Autofocus, PreviewPane, CopyToClipboard and others.

Added bonus: removes the use of jQuery.

In this PR we:

- rewrite the module as ES Module
- update the tests to be ES Module and polyfill
some Javascript methods that JSDOM doesn't
return correct values
- remove old module JS file
- remove old module from the legacy bundle build

- this is module is used on the scheduling of
notifications on the "send" page and
- upload CSV emergency list preview page

kr8n3r committed:
Extend ScreenMock.mockPositionAndDimension method

In certain cases (fullscreen-table test line 395) we need to update the mock of the 'main' element to mimich a page resize event.

To avoid getting

```
 throw new Error(`An element called '${itemName}' already has its position and dimension mocked`);
 ```
 a new argument called `reMock` has been added with a default `false` value, so that `true` can only be added when needed.

tombye reviewed (approved):
All looks good now. Thanks for all the work on this, it's been a lot! 🎉 🏅 👍🏻
